### PR TITLE
refactor: extract helpers from the connector, clean up the code

### PR DESCRIPTION
**Pull request by vursen**
## Description

- Extracted several helper functions into a separate file `helpers.js` to make the TimePicker connector's source code more lightweight and easier to read.
- Removed unnecessary complexity, e.g. caching the `toLocaleTimeString` options.
- Moved both `helpers.js` and `timepickerConnector.js` to the `vaadin-time-picker` sub-folder (see [the reason](https://github.com/vaadin/flow-components/pull/3049#discussion_r857338642)).

Depends on #3044

## Type of change

- [x] Internal

## Checklist

- [x] I have read the contribution guide: https://vaadin.com/docs-beta/latest/guide/contributing/overview/
- [x] I have added a description following the guideline.
- [x] The issue is created in the corresponding repository and I have referenced it.
- [x] I have added tests to ensure my change is effective and works as intended.
- [x] New and existing tests are passing locally with my change.
- [x] I have performed self-review and corrected misspellings.

### CODE REVIEW

File: vaadin-time-picker-flow-parent/vaadin-time-picker-flow/src/main/resources/META-INF/resources/frontend/timepickerConnector.js [hunk 1]
```diff
@@ -133,112 +75,85 @@
...
+              // thus need to format the time object in correct granularity by passing the format options
+              let localeTimeString = timeToBeFormatted.toLocaleTimeString(locale, {
+                hour: 'numeric',
+                minute: 'numeric',
+                second: includeSeconds() ? 'numeric' : undefined
+              });
+
+              // milliseconds not part of the time format API
+              if (includeMilliSeconds()) {
+                localeTimeString = formatMilliseconds(localeTimeString, timeObject.milliseconds, amString, pmString);
```
Comment (vursen) [inline vaadin-time-picker-flow-parent/vaadin-time-picker-flow/src/main/resources/META-INF/resources/frontend/timepickerConnector.js:100]: **note:** Previously, the `includeMilliSeconds` check was done inside `formatMilliseconds`.

File: vaadin-time-picker-flow-parent/vaadin-time-picker-flow/src/main/resources/META-INF/resources/frontend/timepickerConnector.js [hunk 2]
```diff
@@ -23,73 +32,6 @@
...
-          if (millisecondChars.length === 1) {
-            millisecondChars += '00';
-          } else if (millisecondChars.length === 2) {
-            millisecondChars += '0';
-          }
-          return parseInt(millisecondChars);
-        };
-
-        // detecting milliseconds from input, expects am/pm removed from end, eg. .0 or .00 or .000
-        const millisecondRegExp = /[[\.][\d\u0660-\u0669]{1,3}$/;
```
Comment (vursen) [inline vaadin-time-picker-flow-parent/vaadin-time-picker-flow/src/main/resources/META-INF/resources/frontend/timepickerConnector.js:91]: **note:** This variable was moved [below](https://github.com/vaadin/flow-components/pull/3049/files?diff=split&w=1#diff-fd7ef2beaf5cba9e6bbd9ac707c7d34baec9fe9666353d069e9b20b6de8e5b53R136-R137), closer to the line where it is actually used.

File: vaadin-time-picker-flow-parent/vaadin-time-picker-flow/src/main/resources/META-INF/resources/frontend/helpers.js [hunk 3]
```diff
@@ -0,0 +1,123 @@
+// map from unicode eastern arabic number characters to arabic numbers
```
Comment (sissbruecker) [inline vaadin-time-picker-flow-parent/vaadin-time-picker-flow/src/main/resources/META-INF/resources/frontend/helpers.js:1]: When building the frontend for an application, Flow will collect all sources from all `resources/frontend` folders, and collect them in a single folder. That means that adding  a generic file like `helpers.js` could conflict with sources from other JARs or application code. I would suggest to move this into a component specific folder like `vaadin-time-picker`. See how the connector files for the Map component are structured for example.

**Review by sissbruecker — CHANGES_REQUESTED** [review]

File: vaadin-time-picker-flow-parent/vaadin-time-picker-flow/src/main/resources/META-INF/resources/frontend/helpers.js (hunk 3 above)
Comment (vursen) [reply vaadin-time-picker-flow-parent/vaadin-time-picker-flow/src/main/resources/META-INF/resources/frontend/helpers.js]: Oh, good that you spotted that, thanks. 👍  Moved both `helpers.js` and `timepickerConnector.js` to the `vaadin-time-picker` sub-folder.

**Review by sissbruecker — APPROVED** [review]

File: vaadin-time-picker-flow-parent/vaadin-time-picker-flow/src/main/resources/META-INF/resources/frontend/vaadin-time-picker/helpers.js [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+  // TODO when more scripts support is added (than Arabic), need to exclude those numbers too
+  const amOrPmRegExp = /[^\d\u0660-\u0669]/;
+
+  const matches =
+    // In most locales, the time ends with AM/PM:
+    testTimeString.match(new RegExp(`${amOrPmRegExp.source}+$`, 'g')) ||
+    // In some locales, the time starts with AM/PM e.g in Chinese:
+    testTimeString.match(new RegExp(`^${amOrPmRegExp.source}`, 'g'));
+
+  return matches ? matches[0].trim() : null;
```
Comment (vursen) [inline vaadin-time-picker-flow-parent/vaadin-time-picker-flow/src/main/resources/META-INF/resources/frontend/vaadin-time-picker/helpers.js:58]:
```diff
- const endWithAmPmRegex = /[^\d\u0660-\u0669]+$/g;
- let amPmString = testTimeString.match(endWithAmPmRegex);
- if (!amPmString) {
-   // eg. chinese (and some else too) starts with am/pm
-   amPmString = testTimeString.match(/^[^\d\u0660-\u0669]+/g);
- }
- if (amPmString) {
-   amPmString = amPmString[0].trim();
- }
- return amPmString;
+ const amOrPmRegExp = /[^\d\u0660-\u0669]/;
+
+ const matches =
+   // In most locales, the time ends with AM/PM:
+   testTimeString.match(new RegExp(`${amOrPmRegExp.source}+$`, 'g')) ||
+   // In some locales, the time starts with AM/PM e.g in Chinese:
+   testTimeString.match(new RegExp(`^${amOrPmRegExp.source}`, 'g'));
+ 
+ return matches ? matches[0].trim() : null;
```

File: vaadin-time-picker-flow-parent/vaadin-time-picker-flow/src/main/resources/META-INF/resources/frontend/vaadin-time-picker/helpers.js [hunk 5]
```diff
@@ -0,0 +1,165 @@
...
+  // TODO when more scripts support is added (than Arabic), need to exclude those numbers too
+  const amOrPmRegExp = /[^\d\u0660-\u0669]/;
+
+  const matches =
+    // In most locales, the time ends with AM/PM:
+    testTimeString.match(new RegExp(`${amOrPmRegExp.source}+$`, 'g')) ||
+    // In some locales, the time starts with AM/PM e.g in Chinese:
+    testTimeString.match(new RegExp(`^${amOrPmRegExp.source}`, 'g'));
+
+  return matches && matches[0].trim();
```
Comment (vursen) [inline vaadin-time-picker-flow-parent/vaadin-time-picker-flow/src/main/resources/META-INF/resources/frontend/vaadin-time-picker/helpers.js:58]: **note:**

```diff
- const endWithAmPmRegex = /[^\d\u0660-\u0669]+$/g;
- let amPmString = testTimeString.match(endWithAmPmRegex);
- if (!amPmString) {
-   // eg. chinese (and some else too) starts with am/pm
-   amPmString = testTimeString.match(/^[^\d\u0660-\u0669]+/g);
- }
- if (amPmString) {
-   amPmString = amPmString[0].trim();
- }
- return amPmString;
```
```diff
+ const amOrPmRegExp = /[^\d\u0660-\u0669]/;
+
+ const matches =
+   // In most locales, the time ends with AM/PM:
+   testTimeString.match(new RegExp(`${amOrPmRegExp.source}+$`, 'g')) ||
+   // In some locales, the time starts with AM/PM e.g in Chinese:
+   testTimeString.match(new RegExp(`^${amOrPmRegExp.source}`, 'g'));
+ 
+ return matches && matches[0].trim();
```

File: vaadin-time-picker-flow-parent/vaadin-time-picker-flow/src/main/resources/META-INF/resources/frontend/vaadin-time-picker/helpers.js [hunk 6]
```diff
@@ -0,0 +1,165 @@
...
+  // TODO when more scripts support is added (than Arabic), need to exclude those numbers too
+  const amOrPmRegExp = /[^\d\u0660-\u0669]/;
+
+  const matches =
+    // In most locales, the time ends with AM/PM:
+    testTimeString.match(new RegExp(`${amOrPmRegExp.source}+$`, 'g')) ||
+    // In some locales, the time starts with AM/PM e.g in Chinese:
+    testTimeString.match(new RegExp(`^${amOrPmRegExp.source}+`, 'g'));
+
+  return matches && matches[0].trim();
```
Comment (vursen) [inline vaadin-time-picker-flow-parent/vaadin-time-picker-flow/src/main/resources/META-INF/resources/frontend/vaadin-time-picker/helpers.js:58]: **note:**

```diff
- const endWithAmPmRegex = /[^\d\u0660-\u0669]+$/g;
- let amPmString = testTimeString.match(endWithAmPmRegex);
- if (!amPmString) {
-   // eg. chinese (and some else too) starts with am/pm
-   amPmString = testTimeString.match(/^[^\d\u0660-\u0669]+/g);
- }
- if (amPmString) {
-   amPmString = amPmString[0].trim();
- }
- return amPmString;
```
```diff
+ const amOrPmRegExp = /[^\d\u0660-\u0669]/;
+
+ const matches =
+   // In most locales, the time ends with AM/PM:
+   testTimeString.match(new RegExp(`${amOrPmRegExp.source}+$`, 'g')) ||
+   // In some locales, the time starts with AM/PM e.g in Chinese:
+   testTimeString.match(new RegExp(`^${amOrPmRegExp.source}+`, 'g'));
+ 
+ return matches && matches[0].trim();
```

**Review by sissbruecker — APPROVED** [review]

**Pull request merged**